components/config-mgmt-service/grpcserver/node_export.go

```diff
@@ -71,8 +71,14 @@ func (s *CfgMgmtServer) NodeExport(request *pRequest.NodeExport, stream service.
 	if err != nil {
 		return err
 	}
-
-	return s.exportNodes(stream.Context(), request, exporter)
+	streamCtx := stream.Context()
+	deadline, ok := streamCtx.Deadline()
+	if !ok {
+		deadline = time.Now().Add(time.Minute)
+	}
+	ctx, cancel := context.WithDeadline(streamCtx, deadline)
```

Comment:
[💭 follow-up] We've used this pattern in a few places, I wonder if something like `WithDefaultTimeout(ctx, defaultTimeout)` might be a useful library function that basically does lines 74-79.

Reply:
Hmm counter-proposal:

Let's use [`context.WithTimeout`](https://golang.org/pkg/context/#WithTimeout), it is a simple method [doing the same thing](https://golang.org/src/context/context.go?s=14162:14239#L440). Also, [`context.WithDeadline`](https://golang.org/src/context/context.go?s=12173:12241#L373) already checks if the parent context has a deadline that is before the one we've passed, and ignores ours if that is the case. So, we can just do this:

```go
ctx := stream.Context()
ctx, cancel := context.WithTimeout(ctx, time.Minute)
defer cancel()
```

...and that doesn't warrant a new helper method. 😉

Reply:
@srenatus I thought that at first too, but it is slightly different semantics.  Namely, your formulation will always produce a ctx that will time out in 1 minute.  The formulation in the PR will only produce such a timeout if the existing context doesn't already have a timeout.

Reply:
> Namely, your formulation will always produce a ctx that will time out in 1 minute.

Yup, and that only matters if the calling context has a timeout _longer than one minute_. :+1: If we need this functionality, sure, let's do it 😄

Reply:
What I was trying to do here was add a timeout only if there was not one already. This would allow for the gateway, load balancer, or clients to override the timeout.